File: pc/plugins/variable.py
import re
from core.plugin import Plugin


class VariablePlugin(Plugin):
    def translate(self, line):
        match = re.fullmatch(
            r'変数「([^」]+)」に入れる\s*(.+)',
            line
        )

        if not match:
            return None

        name = match.group(1)
        value = match.group(2)

        # 「文字」→ "文字"
        if value.startswith("「") and value.endswith("」"):
            value = repr(value[1:-1])

        return f"{name} = {value}"

File: web/plugins/variable.js
import { WebPlugin } from "../engine/plugin.js";

export class VariablePlugin extends WebPlugin {
    constructor() {
        super("変数");
    }

    async execute(line, context) {
        const match = line.match(
            /^変数「([^「」]+)」に入れる(?:\s*)(.+)$/
        );

        if (!match) {
            throw new Error(
                `変数の書式が正しくありません: ${line}`
            );
        }

        const [, name, expression] = match;

        const value = context.parser.parseValue(
            expression.trim(),
            context.runtime
        );

        context.runtime.set(name, value);
    }
}